build_projects/dotnet-host-build/CompileTargets.cs

```diff
@@ -180,19 +176,49 @@ public static BuildTargetResult CompileCoreHost(BuildTargetContext c)
 
+                if (configuration.Equals("Release"))
+                {
+                    // Cmake calls it "RelWithDebInfo" in the generated MSBuild
+                    configuration = "RelWithDebInfo";
+                }
+
                 // Why does Windows directly call cmake but Linux/Mac calls "build.sh" in the corehost dir?
                 // See the comment in "src/corehost/build.sh" for details. It doesn't work for some reason.
-                var visualStudio = IsWinx86 ? "Visual Studio 14 2015" : "Visual Studio 14 2015 Win64";
-                var archMacro = IsWinx86 ? "-DCLI_CMAKE_PLATFORM_ARCH_I386=1" : "-DCLI_CMAKE_PLATFORM_ARCH_AMD64=1";
-                var ridMacro = $"-DCLI_CMAKE_RUNTIME_ID:STRING={rid}";
-                var arch = IsWinx86 ? "x86" : "x64";
-                var baseSupportedRid = $"win7-{arch}";
-                var cmakeHostVer = $"-DCLI_CMAKE_HOST_VER:STRING={hostVersion.LatestHostVersion.ToString()}";
-                var cmakeHostPolicyVer = $"-DCLI_CMAKE_HOST_POLICY_VER:STRING={hostVersion.LatestHostPolicyVersion.ToString()}";
-                var cmakeHostFxrVer = $"-DCLI_CMAKE_HOST_FXR_VER:STRING={hostVersion.LatestHostFxrVersion.ToString()}";
-                var cmakeBaseRid = $"-DCLI_CMAKE_PKG_RID:STRING={baseSupportedRid}";
-                var cmakeCommitHash = $"-DCLI_CMAKE_COMMIT_HASH:STRING={commitHash}";
-                var cmakeResourceDir = $"-DCLI_CMAKE_RESOURCE_DIR:STRING={resourceDir}";
+                string cmakeBaseRid, visualStudio, archMacro, arch;
+                string ridMacro = $"-DCLI_CMAKE_RUNTIME_ID:STRING={rid}";
+                string cmakeHostVer = $"-DCLI_CMAKE_HOST_VER:STRING={hostVersion.LatestHostVersion.ToString()}";
+                string cmakeHostPolicyVer = $"-DCLI_CMAKE_HOST_POLICY_VER:STRING={hostVersion.LatestHostPolicyVersion.ToString()}";
+                string cmakeHostFxrVer = $"-DCLI_CMAKE_HOST_FXR_VER:STRING={hostVersion.LatestHostFxrVersion.ToString()}";
+                string cmakeCommitHash = $"-DCLI_CMAKE_COMMIT_HASH:STRING={commitHash}";
+                string cmakeResourceDir = $"-DCLI_CMAKE_RESOURCE_DIR:STRING={resourceDir}";
+
+                switch (platform.ToLower())
+                {
+                    case "x86":
+                        cmakeBaseRid = "-DCLI_CMAKE_PKG_RID:STRING=win7-x86";
+                        visualStudio = "Visual Studio 14 2015";
+                        archMacro = "-DCLI_CMAKE_PLATFORM_ARCH_I386=1";
+                        arch = "x86";
+                        break;
+                    case "arm64":
+                        cmakeBaseRid = "-DCLI_CMAKE_PKG_RID:STRING=win10-arm64";
+                        visualStudio = "Visual Studio 14 2015 Win64";
+                        archMacro = "-DCLI_CMAKE_PLATFORM_ARCH_ARM64=1";
+                        arch = "arm64";
+                        break;
+                    case "x64":
+                    default:
```

Comment:
I would suggest that the `default` case should raise a `PlatformNotSupported` exception so that we do not endup compiling x64 for arm32 when that is brought up.